endpoints/openrtb2/video_auction.go

```diff
@@ -0,0 +1,529 @@
+			err = fmt.Errorf("Invalid JSON in Default Request Settings: %s", err)
+			errL = []error{err}
+			return
+		}
+	}
+
+	//create full open rtb req from full video request
+	mergeData(videoBidReq, bidReq)
+
+	//create impressions array
+	imps, errs := deps.createImpressions(videoBidReq)
+	if errs != nil {
+		errL = append(errL, errs...)
+		handleError(labels, w, errL, ao)
+		return
+	}
+	bidReq.Imp = imps
+	bidReq.ID = "bid_id" //TODO: look at prebid.js
+
+	// Populate any "missing" OpenRTB fields with info from other sources, (e.g. HTTP request headers).
+	deps.setFieldsImplicitly(r, bidReq) // move after merge
+
+	errL = deps.validateRequest(bidReq)
+	if len(errL) > 0 {
+		handleError(labels, w, errL, ao)
+		return
+	}
+
+	ctx := context.Background()
+	cancel := func() {}
+	timeout := deps.cfg.AuctionTimeouts.LimitAuctionTimeout(time.Duration(bidReq.TMax) * time.Millisecond)
+	if timeout > 0 {
+		ctx, cancel = context.WithDeadline(ctx, start.Add(timeout))
+	}
+	defer cancel()
+
+	usersyncs := usersync.ParsePBSCookieFromRequest(r, &(deps.cfg.HostCookie))
+	if bidReq.App != nil {
+		labels.Source = pbsmetrics.DemandApp
+	} else {
+		labels.Source = pbsmetrics.DemandWeb
+		if usersyncs.LiveSyncCount() == 0 {
+			labels.CookieFlag = pbsmetrics.CookieFlagNo
+		} else {
+			labels.CookieFlag = pbsmetrics.CookieFlagYes
+		}
+	}
+
+	numImps = len(bidReq.Imp)
+
+	//execute auction logic
+	response, err := deps.ex.HoldAuction(ctx, bidReq, usersyncs, labels, &deps.categories)
+	ao.Request = bidReq
+	ao.Response = response
+	if err != nil {
+		errL := []error{err}
+		handleError(labels, w, errL, ao)
+		return
+	}
+
+	//build simplified response
+	bidResp, err := buildVideoResponse(response)
+	if err != nil {
+		errL := []error{err}
+		handleError(labels, w, errL, ao)
+		return
+	}
+	if bidReq.Test == 1 {
+		bidResp.Ext = response.Ext
+	}
+
+	resp, err := json.Marshal(bidResp)
+	//resp, err := json.Marshal(response)
+	if err != nil {
+		errL := []error{err}
+		handleError(labels, w, errL, ao)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(resp)
+
+}
+
+func handleError(labels pbsmetrics.Labels, w http.ResponseWriter, errL []error, ao analytics.AuctionObject) {
+	labels.RequestStatus = pbsmetrics.RequestStatusErr
+	w.WriteHeader(http.StatusInternalServerError)
+	var errors string
+	for _, er := range errL {
+		errors = fmt.Sprintf("%s %s", errors, er.Error())
+	}
+	fmt.Fprintf(w, "Critical error while running the video endpoint: %v", errors)
+	glog.Errorf("/openrtb2/video Critical error: %v", errors)
+	ao.Status = http.StatusInternalServerError
+	ao.Errors = append(ao.Errors, errL...)
+}
+
+func (deps *endpointDeps) createImpressions(videoReq *openrtb_ext.BidRequestVideo) (imps []openrtb.Imp, errL []error) {
+	videoDur := videoReq.PodConfig.DurationRangeSec
+	minDuration, maxDuration := minMax(videoDur)
+	reqExactDur := videoReq.PodConfig.RequireExactDuration
+	videoData := videoReq.Video
```

Comment:
Perhaps my comment below about checking for minDuration to be non-zero wasn't well placed.  Assuming it's possible for a zero value to be in videoDur, we can get a divide by zero panic at line 252.  I suggest checking for z value here and returning an error if it is found.

Reply:
Let's avoid the panic by making sure we do not divide by zero. However we should do a better cleanup on the incoming request for allowed duration array in the next phase.

Reply:
Validation added